### PR TITLE
zhushi

### PR DESCRIPTION
**Pull request by small-chen-wen**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Checklist
**Before PR**:

- [ ] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 repos:
   - repo: https://github.com/cpp-linter/cpp-linter-hooks
-    rev: v0.2.1  # Use the ref you want to point at
+    rev: v0.2.4  # Use the ref you want to point at
```
Comment (xintian-514) [inline .pre-commit-config.yaml:3]: 这里为什么要改？

File: DIOPI-PROTO/include/diopi/functions.h [hunk 2]
```diff
@@ -151,10 +151,30 @@ DIOPI_API diopiError_t diopiHardswish(diopiContextHandle_t ctx, diopiTensorHandl
 DIOPI_API diopiError_t diopiHardswishInp(diopiContextHandle_t ctx, diopiTensorHandle_t input);
 DIOPI_API diopiError_t diopiHardswishBackward(diopiContextHandle_t ctx, diopiTensorHandle_t grad_input, diopiConstTensorHandle_t grad_output,
                                               diopiConstTensorHandle_t input);
-
+/**
+ * @brief The function thresholds the input tensor by setting elements greater than a given threshold to the threshold value, while leaving elements less than
+ * or equal to the threshold unchanged.
+ * @param[in] ctx Context environment.
+ * @param input the input tensor. type = [float16,float32, float64].
```
Comment (xintian-514) [inline DIOPI-PROTO/include/diopi/functions.h:158]: 空格，下面都是

File: DIOPI-PROTO/include/diopi/functions.h [hunk 3]
```diff
@@ -151,10 +151,30 @@ DIOPI_API diopiError_t diopiHardswish(diopiContextHandle_t ctx, diopiTensorHandl
...
                                       const diopiScalar_t* value);
+/**
+ * @brief the in-place version of diopiThreshold().
+ * @param[in] ctx Context environment.
+ * @param input the input tensor and will be stored result tensor. type = [float16,float32, float64].
+ * @sa Other parameters refer to diopiThreshold().
+ */
 DIOPI_API diopiError_t diopiThresholdInp(diopiContextHandle_t ctx, diopiTensorHandle_t input, const diopiScalar_t* threshold, const diopiScalar_t* value);
+/**
+ * @brief compute the backward pass of diopiThresholdInp().
```
Comment (xintian-514) [inline DIOPI-PROTO/include/diopi/functions.h:173]: diopiThresholdInp

File: DIOPI-PROTO/include/diopi/functions.h [hunk 4]
```diff
@@ -935,11 +973,32 @@ DIOPI_API diopiError_t diopiLogicalNot(diopiContextHandle_t ctx, diopiTensorHand
...
+ * @brief Computes the bitwise AND of the given input tensors.
+ * @param[in] ctx Context environment.
+ * @param input the first tensor. type = [int16, int32, int64, int8, uint8, bool].
+ * @param other the second tesnor. type = [int16, int32, int64, int8, uint8, bool].
+ * @param[out] out the output tensor. type = [int16, int32, int64, int8, uint8, bool].
  */
 DIOPI_API diopiError_t diopiBitwiseAnd(diopiContextHandle_t ctx, diopiTensorHandle_t out, diopiConstTensorHandle_t input, diopiConstTensorHandle_t other);
+/**
+ * @brief the in-place version of diopiBitwiseAnd().
+ * @param[in] input the input tensor and will be stored result tensor.
```
Comment (xintian-514) [inline DIOPI-PROTO/include/diopi/functions.h:985]: 类型

File: DIOPI-PROTO/include/diopi/functions.h [hunk 5]
```diff
@@ -935,11 +973,32 @@ DIOPI_API diopiError_t diopiLogicalNot(diopiContextHandle_t ctx, diopiTensorHand
...
+ * @brief Computes the bitwise AND of the given input tensors.
+ * @param[in] ctx Context environment.
+ * @param input the first tensor. type = [int16, int32, int64, int8, uint8, bool].
+ * @param other The scalar value to be bitwise and. type = [int16, int32, int64, int8, uint8, bool].
+ * @param[out] out the output tensor. type = [int16, int32, int64, int8, uint8, bool].
+ */
 DIOPI_API diopiError_t diopiBitwiseAndScalar(diopiContextHandle_t ctx, diopiTensorHandle_t out, diopiConstTensorHandle_t input, const diopiScalar_t* other);
+/**
+ * @brief the in-place version of diopiBitwiseAndScalar().
+ * @param[in] input the input tensor and will be stored result tensor.
```
Comment (xintian-514) [inline DIOPI-PROTO/include/diopi/functions.h:999]: 类型

File: DIOPI-PROTO/include/diopi/functions.h [hunk 6]
```diff
@@ -1181,6 +1245,12 @@ DIOPI_API diopiError_t diopiStd(diopiContextHandle_t ctx, diopiTensorHandle_t ou
  */
 DIOPI_API diopiError_t diopiMin(diopiContextHandle_t ctx, diopiTensorHandle_t min, diopiTensorHandle_t min_indices, diopiConstTensorHandle_t input,
                                 int64_t dim);
+/**
```
Comment (xintian-514) [inline DIOPI-PROTO/include/diopi/functions.h:1248]: 上面diopiMin是不是有点问题，应该是某个维度的？

File: DIOPI-PROTO/include/diopi/functions.h [hunk 7]
```diff
@@ -1193,6 +1263,12 @@ DIOPI_API diopiError_t diopiMinAll(diopiContextHandle_t ctx, diopiTensorHandle_t
  */
```
Comment (xintian-514) [inline DIOPI-PROTO/include/diopi/functions.h:1263]: 同上

File: DIOPI-PROTO/include/diopi/functions.h [hunk 8]
```diff
@@ -1257,16 +1333,43 @@ DIOPI_API diopiError_t diopiIndex(diopiContextHandle_t ctx, diopiTensorHandle_t*
...
+ */
 DIOPI_API diopiError_t diopiIndexSelectBackward(diopiContextHandle_t ctx, diopiTensorHandle_t grad_input, diopiConstTensorHandle_t grad,
                                                 diopiSize_t input_sizes, int64_t dim, diopiConstTensorHandle_t index);
 
 /**
- * \brief Slices the input tensor along the selected dimension at the given index.
+ * @brief Slices the input tensor along the selected dimension at the given index.
+ * @param[in] ctx Context environment.
+ * @param input the input tensor, type = [int32, int16, int64, uint8, int8, bool, float32, float64, float16].
+ * @param dim the dimension along which to slice. type = [int32, int64].
```
Comment (xintian-514) [inline DIOPI-PROTO/include/diopi/functions.h:1361]: dim不就是int64_t么

File: DIOPI-PROTO/include/diopi/functions.h [hunk 9]
```diff
@@ -1352,28 +1455,44 @@ DIOPI_API diopiError_t diopiSgd(diopiContextHandle_t ctx, diopiTensorHandle_t w,
...
                                       int64_t padding_idx, bool scale_grad_byfreq, bool sparse);
 DIOPI_API diopiError_t diopiEmbeddingBackward(diopiContextHandle_t ctx, diopiTensorHandle_t out, diopiConstTensorHandle_t grad,
                                               diopiConstTensorHandle_t indices, int64_t num_weights, int64_t padding_idx, bool scale_grad_byfreq, bool sparse);
 
 /**
- * \brief Returns the lower triangular part of the matrix (2-D tensor) or batch of matrices input.
+ * @brief Returns the lower triangular part of the matrix (2-D tensor) or batch of matrices input.
+ * @param[in] ctx Context environment.
+ * @param input the input tensor. type = [float32, float64, float16, int16, int32,int64, uint8, int8, bool].
+ * @param diagonal the diagonal to consider. type = [int32, int64].
```
Comment (xintian-514) [inline DIOPI-PROTO/include/diopi/functions.h:1484]: 同上

File: DIOPI-PROTO/include/diopi/functions.h [hunk 10]
```diff
@@ -1352,28 +1455,44 @@ DIOPI_API diopiError_t diopiSgd(diopiContextHandle_t ctx, diopiTensorHandle_t w,
...
  */
 DIOPI_API diopiError_t diopiTril(diopiContextHandle_t ctx, diopiTensorHandle_t out, diopiConstTensorHandle_t input, int64_t diagonal);
 
 /**
- * \brief Concatenates the given sequence of seq tensors in the given dimension.
+ * @brief Concatenates the given sequence of seq tensors in the given dimension.
+ * @param[in] ctx Context environment.
+ * @param tensors the list of the input tensor list. type = [float32, float16, float64, int16, int64, uint8, int8, bool, int32].
+ * @param num_inputs the number of input tensor list.
+ * @param dim the dimension over which the tensors are concatenated. type = [int32, int64].
```
Comment (xintian-514) [inline DIOPI-PROTO/include/diopi/functions.h:1494]: 同上

File: DIOPI-PROTO/include/diopi/functions.h [hunk 11]
```diff
@@ -1598,9 +1717,14 @@ DIOPI_API diopiError_t diopiCdistBackward(diopiContextHandle_t ctx, diopiTensorH
                                           diopiConstTensorHandle_t input1, diopiConstTensorHandle_t input2, double p, diopiConstTensorHandle_t cdist);
 
 /**
- * \brief Returns the indices of the maximum values of a tensor across a dimension.
+ * @brief Returns the indices of the maximum values of a tensor across a dimension.
+ * @param[in] ctx Context environment.
+ * @param input the input tensor. type=[float32, float64, float16, int16, int32, int64, uint8, int8, bool].
+ * @param dim the dimension to do the operation over. type=[int32, int64].
```
Comment (xintian-514) [inline DIOPI-PROTO/include/diopi/functions.h:1723]: 同上

File: DIOPI-PROTO/include/diopi/functions.h [hunk 12]
```diff
@@ -1598,9 +1717,14 @@ DIOPI_API diopiError_t diopiCdistBackward(diopiContextHandle_t ctx, diopiTensorH
...
+ * @param keepdim whether the output tensor has dim retained or not. type = [bool].
```
Comment (xintian-514) [inline DIOPI-PROTO/include/diopi/functions.h:1724]: 这也不用标类型了

File: DIOPI-PROTO/include/diopi/functions.h [hunk 13]
```diff
@@ -1598,9 +1717,14 @@ DIOPI_API diopiError_t diopiCdistBackward(diopiContextHandle_t ctx, diopiTensorH
...
+ * @param[out] out the output tensor. type=[int32, int64].
  */
-DIOPI_API diopiError_t diopiArgmax(diopiContextHandle_t ctx, diopiTensorHandle_t out, diopiConstTensorHandle_t input, const int64_t* dim, bool keepdim);
+* / DIOPI_API diopiError_t diopiArgmax(diopiContextHandle_t ctx, diopiTensorHandle_t out, diopiConstTensorHandle_t input, const int64_t* dim, bool keepdim);
```
Comment (xintian-514) [inline DIOPI-PROTO/include/diopi/functions.h:1727]: 删掉前面

File: DIOPI-PROTO/include/diopi/functions.h [hunk 14]
```diff
@@ -1694,12 +1818,20 @@ DIOPI_API diopiError_t diopiIndexFillInp(diopiContextHandle_t ctx, diopiTensorHa
...
  */
 DIOPI_API diopiError_t diopiExpand(diopiContextHandle_t ctx, diopiTensorHandle_t out, diopiConstTensorHandle_t input);
 
 /**
- * \brief Creates a one-dimensional tensor of size steps whose values are evenly spaced from start to end, inclusive.
+ * @brief Creates a one-dimensional tensor of size steps whose values are evenly spaced from start to end, inclusive.
+ * @param[in] ctx Context environment.
+ * @param start the starting value for the set of points. type = [float32, float64, float16, int16, int32, int64]
+ * @param end the ending value for the set of points. type = [float32, float64, float16, int16, int32, int64]
+ * @param steps the number of steps to take from start to end. type = [[float32, float64, float16, int32, int64]
```
Comment (xintian-514) [inline DIOPI-PROTO/include/diopi/functions.h:1833]: 类型

File: DIOPI-PROTO/include/diopi/functions.h [hunk 15]
```diff
@@ -1694,12 +1818,20 @@ DIOPI_API diopiError_t diopiIndexFillInp(diopiContextHandle_t ctx, diopiTensorHa
...
+ * @param[in] input the input tensor, type = [float32, float64, float16, int16, int32, int64, uint8, int8, bool]
+ * @param[out] out the output tensor. type = [float32, float64, float16, int16, int32, int64, uint8, int8, bool]
  */
 DIOPI_API diopiError_t diopiExpand(diopiContextHandle_t ctx, diopiTensorHandle_t out, diopiConstTensorHandle_t input);
 
 /**
- * \brief Creates a one-dimensional tensor of size steps whose values are evenly spaced from start to end, inclusive.
+ * @brief Creates a one-dimensional tensor of size steps whose values are evenly spaced from start to end, inclusive.
+ * @param[in] ctx Context environment.
+ * @param start the starting value for the set of points. type = [float32, float64, float16, int16, int32, int64]
```
Comment (xintian-514) [inline DIOPI-PROTO/include/diopi/functions.h:1831]: start和end这些类型都可以是吗

File: DIOPI-PROTO/include/diopi/functions.h [hunk 16]
```diff
@@ -1804,30 +1936,77 @@ DIOPI_API diopiError_t diopiRemainderScalar(diopiContextHandle_t ctx, diopiTenso
 DIOPI_API diopiError_t diopiRemainder(diopiContextHandle_t ctx, diopiTensorHandle_t out, const diopiScalar_t* input, diopiConstTensorHandle_t other);
 
 /**
- * \brief Gathers values along an axis specified by dim.
+ * @brief Gathers values along an axis specified by dim.
+ * @param[in] ctx Context environment.
+ * @param input the input tensor. dtype = [float32, float64, float16, int16, int32, int64, uint8, int8, bool].
+ * @param dim the axis along which to index. dtype = [int32, int64].
```
Comment (xintian-514) [inline DIOPI-PROTO/include/diopi/functions.h:1942]: 类型

File: DIOPI-PROTO/include/diopi/functions.h [hunk 17]
```diff
@@ -1804,30 +1936,77 @@ DIOPI_API diopiError_t diopiRemainderScalar(diopiContextHandle_t ctx, diopiTenso
...
+ * @param[in] input the input and output tensor. dtype = [float32, float64, float16, int16, int32, int64, uint8, int8, bool].
+ * @sa other parameters refer to diopiScatterScalar().
+ */
 DIOPI_API diopiError_t diopiScatterInpScalar(diopiContextHandle_t ctx, diopiTensorHandle_t input, int64_t dim, const diopiScalar_t* value,
                                              diopiConstTensorHandle_t index, const char* reduce);
+/**
+ * @brief Writes all values from the tensor src into input at the indices specified in the index tensor.
+ * @param[in] ctx Context environment.
+ * @param input the input tensor. dtype = [float32, float64, float16, int16, int32, int64, uint8, int8, bool].
+ * @param dim the axis along which to index. dtype = [int32, int64].
```
Comment (xintian-514) [inline DIOPI-PROTO/include/diopi/functions.h:1977]: 类型

File: DIOPI-PROTO/include/diopi/functions.h [hunk 18]
```diff
@@ -1804,30 +1936,77 @@ DIOPI_API diopiError_t diopiRemainderScalar(diopiContextHandle_t ctx, diopiTenso
...
+ * @param src the source tensor. dtype = [float32, float64, float16, int16, int32, int64, uint8, int8, bool].
+ * @param index the indices of elements to scatter. dtype = [int32, int64].
+ * @param reduce the reduce operation. dtype = [string].
+ * @param[out] out the output tensor. dtype = [float32, float64, float16, int16, int32, int64, uint8, int8, bool].
+ */
 DIOPI_API diopiError_t diopiScatter(diopiContextHandle_t ctx, diopiTensorHandle_t out, diopiConstTensorHandle_t input, int64_t dim,
                                     diopiConstTensorHandle_t src, diopiConstTensorHandle_t index, const char* reduce);
+/**
+ * @brief Writes all values from the tensor value into input at the indices specified in the index tensor.
+ * @param[in] value the value to write into input at the indices. dtype = [float32, float64, float16, int16, int32, int64, uint8, int8, bool].
```
Comment (xintian-514) [inline DIOPI-PROTO/include/diopi/functions.h:1987]: value这些类型都支持吗

File: DIOPI-PROTO/include/diopi/functions.h [hunk 19]
```diff
@@ -1804,30 +1936,77 @@ DIOPI_API diopiError_t diopiRemainderScalar(diopiContextHandle_t ctx, diopiTenso
...
  */
 DIOPI_API diopiError_t diopiIndexPutInp(diopiContextHandle_t ctx, diopiTensorHandle_t input, diopiConstTensorHandle_t values, diopiConstTensorHandle_t* indices,
                                         int64_t indices_counts, bool accumulate);
+/**
+ * @brief Puts values from the tensor values into the tensor input using the indices specified in indices.
+ * @param[in] ctx Context environment.
+ * @param input the input tensor. dtype = [float32, float64, float16, int16, int32, int64, uint8, int8, bool].
+ * @param values the tensor containing the values to copy into input. dtype = [float32, float64, float16, int16, int32, int64, uint8, int8, bool].
+ * @param indices the indices into input. dtype = [int32, int64].
+ * @param indices_counts the number of indices. dtype = [int32, int64].
```
Comment (xintian-514) [inline DIOPI-PROTO/include/diopi/functions.h:2006]: 类型

File: DIOPI-PROTO/include/diopi/functions.h [hunk 20]
```diff
@@ -1804,30 +1936,77 @@ DIOPI_API diopiError_t diopiRemainderScalar(diopiContextHandle_t ctx, diopiTenso
...
+ * @param accumulate whether to accumulate into input (if true) or perform a copy (if false). dtype = [bool].
```
Comment (xintian-514) [inline DIOPI-PROTO/include/diopi/functions.h:2007]: bool也不用标了

File: DIOPI-PROTO/include/diopi/functions.h [hunk 21]
```diff
@@ -1346,44 +1451,60 @@ DIOPI_API diopiError_t diopiRoiAlignBackward(diopiContextHandle_t ctx, diopiTens
...
  */
 DIOPI_API diopiError_t diopiClipGradNorm(diopiContextHandle_t ctx, double* out, diopiTensorHandle_t* grads, int64_t num_grads, double max_norm,
                                          double norm_type, bool error_if_nonfinite);
 
-/**
- * \brief A simple lookup table that looks up embeddings in a fixed dictionary and size.
- */
 DIOPI_API diopiError_t diopiEmbeddingRenorm_(diopiContextHandle_t ctx, diopiTensorHandle_t inout, diopiConstTensorHandle_t indices, double max_norm,
                                              double norm_type);
+/**
```
Comment (xintian-514) [inline DIOPI-PROTO/include/diopi/functions.h:1474]: 这个是为啥

**Review by xintian-514 — APPROVED** [review]

File: DIOPI-PROTO/include/diopi/functions.h (hunk 15 above)
Comment (small-chen-wen) [reply DIOPI-PROTO/include/diopi/functions.h:1833]: diopi_config里面没有写。

File: DIOPI-PROTO/include/diopi/functions.h (hunk 21 above)
Comment (small-chen-wen) [reply DIOPI-PROTO/include/diopi/functions.h:1474]: diopiEmbeddingRenorm_和 diopiEmbedding应该有点不一样吧，这个注释是diopiEmbedding的吧

**Pull request merged**